src/super_gradients/training/utils/media/video.py

```diff
@@ -75,7 +98,7 @@ def save_video(output_path: str, frames: List[np.ndarray], fps: int) -> None:
     if check_is_gif(output_path):
         save_gif(output_path, frames, fps)
     else:
-        save_mp4(output_path, frames, fps)
+        lazy_save_mp4(output_path, frames, fps)
```

Comment:
I think you can actually modify exissting function and instead of taking List[[np.ndarray] make it taking Iterable[np.ndarray] which would cover both scenarios (lazy & precomputed) predictions